structure/heap/heap_test.go

```diff
@@ -0,0 +1,75 @@
+package heap
+
+import "testing"
+
+func TestInsert(t *testing.T) {
+	h := New(7)
+	h.Insert(5)
+
+	if h.Max() != 5 {
+		t.Errorf("max element should be 5, but found %d", h.Max())
+	}
+	if h.Size != 1 {
+		t.Errorf("size of heap should be 1, but found %d", h.Size)
+	}
+
+	h.Insert(6)
+	h.Insert(1)
+	h.Insert(9)
+	h.Insert(78)
+
+	if h.Max() != 78 {
+		t.Errorf("max element should be 78, but found %d", h.Max())
+	}
+	if h.Size != 5 {
+		t.Errorf("size of heap should be 5, but found %d", h.Size)
+	}
+}
+
+func TestMax(t *testing.T) {
+	h := prepareHeapForTest()
+	if h.Max() != 78 {
+		t.Errorf("max element should be 78, but found %d", h.Max())
+	}
+}
+
+func TestExtractMax(t *testing.T) {
+	h := prepareHeapForTest()
+
+	max := h.ExtractMax()
+	if max != 78 {
+		t.Errorf("max element should be 78, but found %d", h.Max())
+	}
+	if h.Size != 3 {
+		t.Errorf("size of heap should be 3, but found %d", h.Size)
+	}
+
+	if h.Max() != 9 {
+		t.Errorf("max element should be 9, but found %d", h.Max())
+	}
+}
+
```

Comment:
Could you take a look at your test coverage? Currently you haven't done any tests on `shiftUp` and `shiftDown`. I think they should be tested but if there is a reason why they cannot be tested let us know.